metricshub-jawk-extension/src/main/java/org/sentrysoftware/metricshub/extension/jawk/MetricsHubExtensionForJawk.java

```diff
@@ -0,0 +1,333 @@
+		EXECUTE_WBEM_REQUEST,
+		EXECUTE_WMI_REQUEST,
+		JSON_2CSV
+	};
+
+	private SourceProcessor sourceProcessor;
+	private String hostname;
+
+	@Override
+	public String getExtensionName() {
+		return "MetricsHubExtensionForJawk";
+	}
+
+	@Override
+	public String[] extensionKeywords() {
+		return KEYWORDS;
+	}
+
+	@Override
+	public int[] getAssocArrayParameterPositions(final String extensionKeyword, final int numArgs) {
+		if (
+			extensionKeyword.equals(EXECUTE_IPMI_REQUEST) ||
+			extensionKeyword.equals(EXECUTE_SNMP_GET) ||
+			extensionKeyword.equals(EXECUTE_SNMP_TABLE) ||
+			extensionKeyword.equals(EXECUTE_WBEM_REQUEST) ||
+			extensionKeyword.equals(EXECUTE_WMI_REQUEST) ||
+			extensionKeyword.equals(EXECUTE_HTTP_REQUEST) ||
+			extensionKeyword.equals(JSON_2CSV)
+		) {
+			return new int[] { 0 };
+		} else {
+			return super.getAssocArrayParameterPositions(extensionKeyword, numArgs);
+		}
+	}
+
+	@Override
+	public Object invoke(String keyword, Object[] args) {
+		if (keyword.equals(EXECUTE_HTTP_REQUEST)) {
+			checkNumArgs(args, 1);
+			return executeHttpRequest(args);
+		} else if (keyword.equals(EXECUTE_IPMI_REQUEST)) {
+			checkNumArgs(args, 1);
+			return executeIpmiRequest(args);
+		} else if (keyword.equals(EXECUTE_SNMP_GET)) {
+			checkNumArgs(args, 1);
+			return executeSnmpGetRequest(args);
+		} else if (keyword.equals(EXECUTE_SNMP_TABLE)) {
+			checkNumArgs(args, 1);
+			return executeSnmpTableRequest(args);
+		} else if (keyword.equals(EXECUTE_WBEM_REQUEST)) {
+			checkNumArgs(args, 1);
+			return executeWbemRequest(args);
+		} else if (keyword.equals(EXECUTE_WMI_REQUEST)) {
+			checkNumArgs(args, 1);
+			return executeWmiRequest(args);
+		} else if (keyword.equals(JSON_2CSV)) {
+			checkNumArgs(args, 1);
+			return executeJson2csv(args);
+		} else {
+			throw new NotImplementedError(keyword);
+		}
+	}
+
+	/**
+	 * Execute a HTTP request through the context.
+	 * @param args The array of arguments to use to create the {@link HttpSource}.
+	 * @return The table result from the execution of the {@link HttpSource}.
+	 */
+	private String executeHttpRequest(final Object[] args) {
+		if ((args[0] instanceof AssocArray argsAssocArray)) {
+			return executeSource(
+				HttpSource
+					.builder()
+					.method(HttpMethod.valueOf(toAwkString(argsAssocArray.get("method")).toUpperCase()))
+					.path(toAwkString(argsAssocArray.get("path")))
+					.header(toAwkString(argsAssocArray.get("header")))
+					.body(toAwkString(argsAssocArray.get("body")))
+					.authenticationToken(toAwkString(argsAssocArray.get("authenticationToken")))
+					.resultContent(ResultContent.detect(toAwkString(argsAssocArray.get("resultContent"))))
+					.forceSerialization(toAwkString(argsAssocArray.get("forceSerialization")).equals("true"))
+					.build()
+			);
+		} else {
+			log.warn(
+				"Hostname {} - Jawk Source is invalid, The HTTP request arguments must be provided through an Association array. The Jawk operation will return an empty result",

```

Comment:
```suggestion
				"Hostname {} - Jawk: executeHttpRequest(): Expected Association array.",
```

(and similarly for other error messages)